### PR TITLE
Add new log file structure 

### LINKED ISSUE #597: Log cannot identify sensor type
Perhaps we can put the device type as the first byte in the file for a fast hotfix while we think about our renewing logging requirements and log format.

### PR DESCRIPTION
**Pull request by patrickelectric**
Fix #597

### CODE REVIEW

File: src/devicemanager/devicemanager.cpp [hunk 1]
```diff
@@ -121,6 +121,17 @@ void DeviceManager::connectLinkDirectly(AbstractLinkNamespace::LinkType connType
...
 
+void DeviceManager::playLogFile(AbstractLinkNamespace::LinkType connType, const QStringList& connString)
+{
+    auto linkConfiguration = LinkConfiguration{connType, connString};
+    linkConfiguration.setDeviceType(PingDeviceType::PING360);
+
+    // Append configuration as device of type "None"
+    // This will create and populate all necessary roles before connecting
+    append(linkConfiguration);
+    connectLink(&linkConfiguration);
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply src/devicemanager/devicemanager.cpp]: Since this variable is used both from c++ and qml, the argument should be a pointer.

**Comment by patrickelectric** [comment]
@tcanabrava done.

**Comment by jaxxzer** [comment]
@patrickelectric can you describe here what the essential changes are to the log file structure?

It looks like you have basically added a header to the file describing the build and the device. Is there anything else to be aware of?

**Comment by patrickelectric** [comment]
@jaxxzer 	yes, this creates a header in the log file, but not a simple one. This header allows us to be future compatible with changes and updates in the log structure, making any change and improvements in the log file/structure detectable and easy to track in software.

**Comment by patrickelectric** [comment]
@Williangalvani 	if you can, please test.

**Comment by Williangalvani** [comment]
> @Williangalvani if you can, please test.
it saves an loads fine in the tests I did. I tested Ping360 both links and Ping1D via serial.

**Pull request merged**